plasmapy/mathematics/tests/test_dispersion.py

```diff
@@ -1,92 +1,167 @@
-                      atol=atol, rtol=0), \
-        "Symmetry property of Z(zeta*) = -[Z(-zeta)]* not satisfied"
+    assert np.allclose(Z1, Z2, atol=1e-15 * (1 + 1j), rtol=1e-15), \
+        ("plasma_dispersion_func is returning values that are inconsistent "
+         "with the power series expansion given by equation B.3 from Plasma "
+         "Waves by D. G. Swanson (2nd edition, 2003).  The results from "
+         f"plasma_dispersion_func are:\n\n{Z1}\n\n"
+         f"whereas the results from the power series expansion are:\n\n{Z2}\n")
 
-    if zeta.imag > 0:
-        assert np.isclose(plasma_dispersion_func(zeta.conjugate()),
-                          (plasma_dispersion_func(zeta)).conjugate() +
-                          2j*np.sqrt(np.pi)*np.exp(-(zeta.conjugate()**2)),
-                          atol=atol, rtol=0), \
-            "Symmetry property of Z(zeta*) valid for zeta.imag>0 not satisfied"
 
-    assert np.isclose(plasma_dispersion_func(9.2j),
-                      plasma_dispersion_func(9.2j*u.dimensionless_unscaled),
-                      atol=atol, rtol=0)
+def test_plasma_dispersion_func_zeros():
+    """Test the zeros of the plasma dispersion function."""
+    zeros = np.array([
+            1.991_466_842_833_879_6 - 1.354_810_128_112_006_2j,
+            2.691_149_024_251_438_8 - 2.177_044_906_089_615_9j,
+            3.235_330_868_352_816_5 - 2.784_387_613_230_428_2j,
+            3.697_309_702_468_468_4 - 3.287_410_789_389_848_6j,
+            4.106_107_284_682_632_1 - 3.725_948_719_445_790_4j],
+            dtype=np.complex128)
 
-    with pytest.raises(TypeError):
-        plasma_dispersion_func('')
+    for zero in zeros:
+        Z = plasma_dispersion_func(zero)
+        assert np.isclose(Z, 0 + 0j, atol=1e-15 * (1 + 1j)), \
+            ("A zero of the plasma dispersion function is expected at ζ = "
+             f"{zero}, but plasma_dispersion_func({zero}) is equal to {Z}.")
 
-    with pytest.raises(u.UnitsError):
-        plasma_dispersion_func(6*u.m)
 
-    with pytest.raises(ValueError):
-        plasma_dispersion_func(np.inf)
+# ζ, expected
+plasma_disp_deriv_table = [
+    (0, -2),
+    (1, 0.152_318 - 1.304_10j),
+    (1j, -0.484_257),
+    (1.2 + 4.4j, -0.397_561e-1 - 0.217_392e-1j),
+    (9j, plasma_dispersion_func_deriv(9j * u.dimensionless_unscaled)),
+    (5.4 - 3.1j, 0.012_449_1 + 0.023_138_3j),
+    (9.9 - 10j, 476.153 + 553.121j),
+    (5 + 7j, -4.591_20e-3 - 0.012_610_4j),
+    (4.5 - 10j, 0.260_153e37 - 0.211_814e37j)]
 
 
-def test_plasma_dispersion_func_deriv():
-    r"""Test the implementation of plasma_dispersion_func_deriv against
-    tabulated results from Fried & Conte (1961)."""
+@pytest.mark.parametrize('ζ, expected', plasma_disp_deriv_table)
+def test_plasma_dispersion_func_deriv(ζ, expected):
+    r"""Test the implementation of plasma_dispersion_func_deriv
+    against tabulated results and an analytical relationship from
+    Fried & Conte (1961)."""
```

Comment:
Is Fried & Conte a book or a publication? If it's a book then the chapter and page number would be handy. If a publication then the journal name, issue, and article number would be useful.